### PR TITLE
Add XMPlay 3.8.3.4

### PR DESCRIPTION
**Pull request by Retia-Adolf**

### CODE REVIEW

File: bucket/xmplay.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    "bin": "xmplay.exe",
+    "shortcuts": [
+        [
+            "xmplay.exe",
+            "XMPlay"
+        ]
+    ],
+    "pre_install": [
+        "# Create placeholder files",
+        "function CreateFile([String] $file, [String] $content = $null) {",
```
Comment (Ash258) [inline bucket/xmplay.json]:
```suggestion
        "function CreateFile([String[]] $file) {",
        "    foreach ($f in $file) {",
        "        if (! (Test-Path \"$dir\\$f\")) { New-Item \"$dir\\$f\" -ItemType 'File' | Out-Null }",
        "    }",
        "}",
```

File: bucket/xmplay.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        "    if(!(Test-Path \"$persist_dir\\$file\")) {",
+        "        Write-Host \"Create placeholder for xmplay: $file\"",
+        "        New-Item -Force -Path \"$persist_dir\\$file\" -ItemType file -Value $content | Out-Null",
+        "    }",
+        "}",
+        "CreateFile xmplay.ini",
```
Comment (Ash258) [inline bucket/xmplay.json]:
```suggestion
        "CreateFile 'xmplay.ini', 'xmplay.library', 'xmplay.pls'"
```

File: bucket/xmplay.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
+{
+    "version": "3.8.3.4",
+    "license": "freeware",
+    "homepage": "https://www.un4seen.com/xmplay.html",
+    "description": "a Win32 audio player. Initially, when XMPlay was first released in 1998, only the XM format was supported, hence the name \"XMPlay\". Now supporting formats as OGG, MP3, MP2, MP1, WAV, AIFF, CDA, and more with Winamp plugin and its own plugin support.",
+    "url": "https://www.un4seen.com/download.php?xmplay38#/dl.7z",
+    "hash": "sha512:05ccbf67897b4ac55bc1591839ff559b55bd3c2db6aeaf3a4350a71c97485397a6a68fe5a54cf12cc2d51581b944444554d7adc3fe586fcba1195c6f2d4bacbe",
```
Comment (Ash258) [inline bucket/xmplay.json]: Why 512, when there is no hash property in autoupdate?

Run autoupdate.

File: bucket/xmplay.json [hunk 4]
```diff
@@ -0,0 +1,38 @@
+{
+    "version": "3.8.3.4",
+    "license": "freeware",
```
Comment (Ash258) [inline bucket/xmplay.json]:
```suggestion
    "license": "Freeware",
```

File: bucket/xmplay.json [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    "homepage": "https://www.un4seen.com/xmplay.html",
+    "description": "a Win32 audio player. Initially, when XMPlay was first released in 1998, only the XM format was supported, hence the name \"XMPlay\". Now supporting formats as OGG, MP3, MP2, MP1, WAV, AIFF, CDA, and more with Winamp plugin and its own plugin support.",
```
Comment (Ash258) [inline bucket/xmplay.json]: No one need histrorical lesson

```suggestion
    "description": "Win32 audio player with support for Winamp plugins.",
```

File: bucket/xmplay.json (hunk 4 above)
Comment (Ash258) [reply bucket/xmplay.json]: Where did you get this information?

Comment (Retia-Adolf) [reply bucket/xmplay.json]: https://en.wikipedia.org/wiki/XMPlay

File: bucket/xmplay.json [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    "url": "https://www.un4seen.com/files/xmplay38.zip",
+    "hash": "sha512:05ccbf67897b4ac55bc1591839ff559b55bd3c2db6aeaf3a4350a71c97485397a6a68fe5a54cf12cc2d51581b944444554d7adc3fe586fcba1195c6f2d4bacbe",
+    "bin": "xmplay.exe",
+    "shortcuts": [
+        [
+            "xmplay.exe",
+            "XMPlay"
+        ]
+    ],
+    "pre_install": [
```
Comment (Ash258) [inline bucket/xmplay.json]: Move `pre_install` property before bin property

File: bucket/xmplay.json [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+        "        if (! (Test-Path \"$dir\\$f\")) { New-Item \"$dir\\$f\" -ItemType 'File' | Out-Null }",
+        "    }",
+        "}",
+        "CreateFile 'xmplay.ini', 'xmplay.library', 'xmplay.pls'"
+    ],
+    "persist": [
+        "xmplay.ini",
+        "xmplay.library",
+        "xmplay.pls",
+        "user-plugins",
```
Comment (Ash258) [inline bucket/xmplay.json]: Why manifest do not follow this structure?

A

http://support.xmplay.com/article.php?id=45

Comment (Retia-Adolf) [reply bucket/xmplay.json]: Ok, it scans any folder for plugins and skins under same directory where it is.

**Pull request merged**